ophyd/areadetector/filestore_mixins.py

```diff
@@ -46,15 +48,18 @@ def new_short_uid():
     return '-'.join(new_uid().split('-')[:-1])
 
 
-def _ensure_trailing_slash(path):
+def _ensure_trailing_slash(path, path_semantics=None):
     """
     'a/b/c' -> 'a/b/c/'
 
     EPICS adds the trailing slash itself if we do not, so in order for the
     setpoint filepath to match the readback filepath, we need to add the
     trailing slash ourselves.
     """
-    return os.path.join(path, '')
+    if path_semantics is 'posix':
+        return f'{PurePosixPath(path)}/'
+    else:
+        return f'{PurePath(path)}{os.path.sep}'
```

Comment:
Consider the scenario where ophyd is running on a Linux workstation but the area detector driver is running on Windows, as is the case at some beamlines. In that situation, `os.path.sep` will return `/` but the driver needs to be set to `\`. I suggest something along the lines of:

```py
if path_semantics == 'posix':
    return f'{PurePosixPath(path)}/'
elif path_semantics == 'windows':
    return  f'{PurePath(path)}\'
elif path_semantics is None:
    # We are forced to guess which path semantics to use.
    # Guess that the AD driver is running on the same OS as this client.
    return f'{PurePath(path)}{os.path.sep}'
else:
    # This should never happen, but just for the sake of future-proofing....
    raise ValueError(f"Cannot handle path_semantics={path_semantics}")
```

Reply:
@danielballan, I like the more explicit conditions.
For the Windows part it should be either `f'{PurePath(path)}\\\\'` or `fr'{PurePath(path)}\\'` to have `\\` returned at the end. Also, I think it should better be `PureWindowsPath`.

Reply:
I played with it a bit, and `f'{PurePath(path)}\\'` should be OK (we need two backslashes to escape a special backslash symbol).